### DISCUSSION TITLE
Integrating with Microsoft Teams for notifications

### DISCUSSION THREAD

**Discussion by sabreensalama** [argoproj/argo-workflows#12268] [Q&A] [answered]
Hello I hope to find integration between argo-workflows and Microsoft Teams or Slack or Microsoft Outlook once the workflow has failed or with an error or even when succeeded

**Comment by agilgur5** [comment, marked as answer]
You can integrate many types of [`EventSources`](https://argoproj.github.io/argo-events/concepts/event_source/) with [Argo Events](https://github.com/argoproj/argo-events), including [for `Workflow` resources](https://github.com/argoproj/argo-events/blob/69ac700e96c711492cc21d9c88e235db3f7cc902/examples/event-sources/resource.yaml#L8) (or any other k8s resource). There are built-in [`Triggers`](https://argoproj.github.io/argo-events/concepts/trigger/) for some things like Slack notifications.

You can also use Workflows's [Exit Handlers](https://argoproj.github.io/argo-workflows/walk-through/exit-handlers/) feature to perform an action on `Workflow` completion.
